File: src/main/java/com/grepp/synapse4/app/model/user/MailService.java
package com.grepp.synapse4.app.model.user;

import com.grepp.synapse4.app.model.user.mail.request.MailRequest;
import lombok.RequiredArgsConstructor;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.beans.factory.annotation.Qualifier;
import org.springframework.mail.SimpleMailMessage;
import org.springframework.mail.javamail.JavaMailSender;
import org.springframework.stereotype.Service;
import org.springframework.web.reactive.function.client.WebClient;

@Service
@RequiredArgsConstructor
public class MailService {

    private final JavaMailSender mailSender;
    // 밑에 태그는 현재 WebClientConfig 에
    // 3개의 빈이 등록되어 어떤 빈을
    // 호출할지 인식하지못해 명시적으로 선언해놨습니다.
    @Qualifier("mailWebClient")
    private final WebClient mailWebClient;

    // 메일서버로 요청
    public void sendToServerTempPasswordEmail(String toEmail, String tempPassword) {
        MailRequest request = new MailRequest(toEmail, tempPassword);

        mailWebClient.post()
                .uri("/mail/send")// 메일 서버 컨트로럴로 요청
                .bodyValue(request)
                .retrieve()
                .bodyToMono(Void.class)
                .block();
    }
}

File: mail-service/mail-service/src/main/kotlin/com/grepp/spring/app/model/MailService.kt
package com.grepp.spring.app.model

import com.grepp.spring.app.controller.payload.SmtpRequest
import com.grepp.spring.app.model.code.MailTemplatePath
import com.grepp.spring.infra.mail.MailTemplate
import com.grepp.spring.infra.mail.SmtpDto
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch
import kotlinx.coroutines.runBlocking
import kotlinx.coroutines.sync.Semaphore
import kotlinx.coroutines.sync.withPermit
import org.slf4j.LoggerFactory
import org.springframework.mail.SimpleMailMessage
import org.springframework.mail.javamail.JavaMailSender
import org.springframework.stereotype.Service

@Service
class MailService(
    private val mailSender: JavaMailSender
) {
    fun send(request: SmtpRequest){
        val subject = "[밥로드] 임시 비밀번호 안내";
        val content = """요청하신 임시 비밀번호는 아래와 같습니다.
            
            임시 비밀번호: ${request.tempPassword}
            
            로그인 후 반드시 비밀번호를 변경해주세요.
        """.trimIndent()

        val message = SimpleMailMessage().apply {
            setTo(request.toEmail)
            setSubject(subject)
            setText(content)
        }
        mailSender.send(message)
    }
}